tap_github/organization_streams.py

```diff
@@ -211,3 +213,820 @@ class TeamRolesStream(GitHubRestStream):
         th.Property("role", th.StringType),
         th.Property("state", th.StringType),
     ).to_dict()
+
+
+class ProjectsStream(GitHubGraphqlStream):
+    """Fetches GitHub projects (new projects aka ProjectsV2) for an organization.
+
+    API Reference: https://docs.github.com/en/graphql/reference/objects#projectv2
+    """
+
+    name = "projects"
+    primary_keys: ClassVar[list[str]] = ["org", "id"]
+    parent_stream_type = OrganizationStream
+    ignore_parent_replication_key = True
+    state_partitioning_keys: ClassVar[list[str]] = ["org"]
+    query_jsonpath = "$.data.organization.projectsV2.nodes[*]"
+
+    @property
+    def query(self) -> str:
+        """GraphQL query to fetch projects."""
+        return """
+        query OrganizationProjects($org: String!, $nextPageCursor_0: String) {
+          organization(login: $org) {
+            projectsV2(first: 100, after: $nextPageCursor_0) {
+              nodes {
+                closed
+                closed_at: closedAt
+                created_at: createdAt
+                creator {
+                  login
+                  resource_path: resourcePath
+                  url
+                  type: __typename
+                }
+                id: fullDatabaseId
+                node_id: id
+                number
+                owner {
+                  node_id: id
+                  type: __typename
+                }
+                public
+                readme
+                resource_path: resourcePath
+                short_description: shortDescription
+                template
+                title
+                updated_at: updatedAt
+                url
+                viewer_can_close: viewerCanClose
+                viewer_can_reopen: viewerCanReopen
+                viewer_can_update: viewerCanUpdate
+              }
+              pageInfo {
+                hasNextPage_0: hasNextPage
+                endCursor_0: endCursor
+                startCursor_0: startCursor
+              }
+              totalCount
+            }
+          }
+          rateLimit {
+            cost
+          }
+        }
+        """
+
+    def get_child_context(self, record: dict, context: Context | None) -> dict:
+        """Return context for child streams."""
+        new_context = {"project_number": record["number"]}
+        if context:
+            return {**context, **new_context}
+        return new_context
+
+    def post_process(self, row: dict, context: Context | None = None) -> dict:
+        """Post-process a fetched record."""
+        row = super().post_process(row, context)
+        if context:
+            row["org"] = context["org"]
+        return row
+
+    schema = th.PropertiesList(
+        # Parent keys
+        th.Property("org", th.StringType),
+        # Project fields
+        th.Property(
+            "id", th.StringType, required=False
```

Comment:
This is currently functionally equivalent, but it's more semantically adequate IMO. Also, we might decide that `required` doesn't affect whether the field is nullable, only whether it's expected to be present.

```suggestion
            "id", th.StringType, nullable=False
```

Reply:
Thanks, should I switch to use `nullable` field for all nullable fields then, or just for IDs for now?

Reply:
Updated the PR to use nullable for id.

Reply:
Thanks!